packages/node_modules/pouchdb-find/src/adapters/local/abstract-mapper.js

```diff
@@ -98,8 +192,9 @@ function mapper(mapFunDef, emit) {
   // mapFunDef is a list of fields
 
   var fields = Object.keys(mapFunDef.fields);
+  var selector = mapFunDef.partial_filter_selector;
 
-  return createMapper(fields, emit);
+  return createMapper(fields, selector, emit);
```

Comment:
This is nicely done. I'm just wondering if we could remove the amount of code that is copied and pasted from the previous map functions. Would it be possible to do something like this here instead of having different map functions:
```
if (!selector) {
    return createMapper(fields, emit);
} else {
    var mapper = createMapper(fields, emit);
    return function (doc) {
         if (!rowFilter(doc, selector, inMemoryFields)) {
            return;
         }

          return mapper(doc);
    };
}
```